### ISSUE TITLE
question: need to understand process for using root CA within Zalenium grid containers

### ISSUE THREAD

**Issue by borisivan** [zalando/zalenium#486]
Zalenium Image Version(s):  
latest, every execution does a pull first.

Docker Version:  
17.09.0-ce

OS: 
SLES, Ubuntu, others

Docker Command to start Zalenium:  
docker, run, -tid, --name=obfuscated, --net=host, -p, 4444:4444, -v, /var/run/docker.sock:/var/run/docker.sock, -v, /usr/bin/docker:/usr/bin/docker, -v, /dev/shm:/dev/shm, -v, /tmp/videos/obfuscated:/home/seluser/videos, -v, /tmp/obfuscated:/tmp/node/home/seluser/Downloads, --privileged, dosel/zalenium:latest, start, --chromeContainers, 7, --screenWidth, 1920, --seleniumImageName, elgalu/selenium:latest, --screenHeight, 1080, --firefoxContainers, 0, --keepOnlyFailedTests, true, --videoRecordingEnabled, true, --debugEnabled, false, --maxTestSessions, 150

## Expected Behavior -

Browsers utilize https

## Actual Behavior -

Browsers do not utilize https

If I launch a browser manually and specify the webpage via https, it utilizes a secure connection.

Using Zalenium, all browsers revert to http. My remoteWebDriver does a get(URL) with the URL being https. But chrome or FF, both show as unsecured.

This is probably me doing something wrong, not looking to label this as a bug, hopefully you can tag as 'question'. Looking for advice, and if anyone happened to have any example code (java?) re: how they instantiate their remoteWebDriver, I'd appreciate it. I've spent days scouring the internet and there are hundreds of examples, none seem to work for me. I don't think I utilize a proxy, but our company does force us to utilize root certificates, which are imported on the system, so this might be a docker question, re: containers usage of host certificates, etc. Then again, if I use selenium standalone server locally and not Zalenium, I can't get it to work that way, either.

Looking for help.

**Comment by diemol** [comment]
Hi @borisivan,
I just tried myself,
 
Using the browser manually:
![screen shot 2018-03-19 at 16 30 59](https://user-images.githubusercontent.com/5992658/37605192-f839cef2-2b92-11e8-9bfb-3482a39b180c.png)

Running a test with Java:
![screen shot 2018-03-19 at 16 33 23](https://user-images.githubusercontent.com/5992658/37605350-482b2906-2b93-11e8-942b-eae68defd557.png)

So it seems to be related to the root certificates thing, I haven't worked with them but then we might just need to research how to make them available in the container perhaps?

**Comment by borisivan** [comment]
Hi @diemol -- thanks. I feel silly in that I should have been able to prove the same. I tried with selenium server standalone both hub and node launched on my local system, and I'm able to bring up https://www.google.com. Trying with Zalenium, I can't (it reverts to http). As you mention, it's probably that we need to find a way to have the decryption authority or root CA file be accessed (and processed at startup?) by this docker solution.

Extra work required on my end: the internal app I'm really trying to test, won't even load via https when I try w/out Zalenium (i.e. in the way that https://www.google.com does work). But that's likely due to a custom certificate for that app.

Bottom line, yes, I also find that w/out Zalenium, but still using RemoteWebDriver via selenium-server node and hub on this machine, I can bring up a site via https (google). But w/ Zalenium, it reverts to http.

**Comment by borisivan** [comment]
Hi @diemol, I've changed the title accordingly. Agreed that the question/problem is specific to what steps I need to take to have Zalenium use the certificates on the host OS.

**Comment by diemol** [comment]
Hi @borisivan,

When you used Selenium Server standalone, were you using directly the jar file or the docker containers?

Additionally, I was checking a bit, and if there is a root certificate that needs to be used, seems that the certificate must be included inside the docker image (elgalu/selenium in this case). Check this link https://stackoverflow.com/questions/42292444/how-do-i-add-a-ca-root-certificate-inside-a-docker-image

Then maybe you can try to something like:
```
FROM elgalu/selenium
ADD your_ca_root.crt /usr/local/share/ca-certificates/foo.crt
RUN update-ca-certificates
```

and then use the resulting image to run the tests.

**Comment by diemol** [comment]
Oh, seems that you already tried it and succeed, based on this [comment](https://github.com/zalando/zalenium/issues/491#issuecomment-375814549).

I'll close this issue then.

**Comment by borisivan** [comment]
Hi @diemol, agreed, thanks. (and I should have updated this issue myself previously once I found out how to add the root cert to the container and rebuild).

**Comment by akash8663** [comment]
Hi @diemol i have tried this option to copy the crt file in zalenium but still i am getting Connection not private / Not secure issues. Below is the code in my DockerFile. After building i see the message says 1 Added. Also added a screenshot

`FROM elgalu/selenium`
`COPY RootCA01.cer /tmp/ca-certificates/`
`RUN sudo mv /tmp/ca-certificates/RootCA01.cer /usr/local/share/ca-certificates/RootCA01.cer`
`RUN sudo openssl x509 -inform PEM -in /usr/local/share/ca-certificates/RootCA01.cer -out /usr/local/share/ca-certificates/RootCA01.crt`
`RUN sudo chmod 777 /usr/local/share/ca-certificates/RootCA01.crt`
`RUN sudo update-ca-certificates`
`FROM dosel/zalenium`

![image](https://user-images.githubusercontent.com/28462526/78077663-c237e680-7376-11ea-9225-79d8f598fe7b.png)
